Source/Events.Handling/Builder/EventHandlerBuildWarnings.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Dolittle.SDK.Events.Handling.Builder
+{
+    /// <summary>
+    /// Represents the build warnings from when an event handler was attempted to be built.
+    /// </summary>
+    public class EventHandlerBuildWarnings
+    {
+        readonly EventHandlerId _eventHandlerId;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventHandlerBuildWarnings"/> class.
+        /// </summary>
+        /// <param name="eventHandlerId">The <see cref="EventHandlerId" />.</param>
+        /// <param name="warnings">The <see cref="IEnumerable{T}" /> of warning messages.</param>
+        public EventHandlerBuildWarnings(EventHandlerId eventHandlerId, IEnumerable<string> warnings)
+        {
+            _eventHandlerId = eventHandlerId;
+            Warnings = warnings;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventHandlerBuildWarnings"/> class.
+        /// </summary>
+        /// <param name="eventHandlerId">The <see cref="EventHandlerId" />.</param>
+        /// <param name="warnings">The <see cref="IEnumerable{T}" /> of warning messages.</param>
+        public EventHandlerBuildWarnings(EventHandlerId eventHandlerId, params string[] warnings)
+            : this(eventHandlerId, warnings.ToList())
+        {
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IEnumerable{T}" /> of warnings when attempting to build the event handler.
+        /// </summary>
+        public IEnumerable<string> Warnings { get; }
+
+        /// <inheritdoc/>
+        public override string ToString()
+            => $"Failed to build event handler {_eventHandlerId}{System.Environment.NewLine}{string.Join($"{System.Environment.NewLine}\t", Warnings)}";
```

Comment:
Newlines in log messages are not that great if we can avoid them, most systems read things line by line, so if we can avoid them that helps parsing a log. In this example, if you were to filter for warning messages (as this is printed somewhere else), you would se the "Failed to build event handler ...", then not the reasons since they are not warnings (because of newlines).

This is somewhat related to my comment above by using the ILogger directly when warnings occur instead. That would make life a bit easier for devs.

Reply:
This should be resolved by latest commits